File: ezpacking/src/main/java/org/zerock/ezpacking/domain/entity/SearchHistory.java
package org.zerock.ezpacking.domain.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "search_history")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class SearchHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String keyword;

    @Column(nullable = false)
    private Long count;

    @Column(nullable = false)
    private boolean carryOnAllowed;

    @Column(nullable = false)
    private boolean checkedAllowed;

    @Column(length = 500)
    private String restrictions;

    public void incrementCount() {
        this.count++;
    }

    public void updateLastSearchResult(boolean carryOnAllowed, boolean checkedAllowed, String restrictions) {
        this.carryOnAllowed = carryOnAllowed;
        this.checkedAllowed = checkedAllowed;
        this.restrictions = restrictions;
    }
}

File: backend/src/main/kotlin/com/example/apiezpz/search/entity/SearchHistory.kt
package com.example.apiezpz.search.entity

import com.example.apiezpz.auth.entity.User
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
data class SearchHistory(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    
    @Column(nullable = false)
    var category: String = "",
    
    @Column(nullable = false)
    var keyword: String = "",
    
    @Column(nullable = false)
    var searchDate: LocalDateTime = LocalDateTime.now(),
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    var user: User? = null
) {
    @PrePersist
    protected fun onCreate() {
        searchDate = LocalDateTime.now()
    }
} 